src/chaat/model/user.clj

```diff
@@ -0,0 +1,72 @@
+(ns chaat.model.user
+  (:require
+   [chaat.db.user :as db.user]
+   [crypto.password.bcrypt :as bcrypt]
+   [java-time.api :as jt]
+   [next.jdbc.date-time :as dt]
+   [chaat.errors :refer [do-or-error]]))
+
+;; Model layer does validation pertaining to the representation 
+;; of a user: username & password format (length, special characters etc.)
+
+(defn min-length?
+  "Check if string is >= a threshold length"
+  [str threshold]
+  (>= (count str) threshold))
+
+;; Add more restrictions: special characters not allowed, only alphanumeric etc.
+;; Perhaps I can start using the schema library? Or implement something similar myself.
```

Comment:
=> bug tracker

Same for other such comments.

Reply:
Yes, cleaned up.